### PR TITLE
update species.r

### PR DESCRIPTION
**Pull request by DingWB**
update species.r

### CODE REVIEW

File: R/species.r [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+#' @param infer.human whether to use the fraction of negative probes to infer human.
+#' @param topN Top n positive and negative probes used to infer species.
+#' @return a list of auc, pvalue, species (NCBI official species names) and taxid.
+#' We infer species based on probes pvalues and alignment score.
+#' AUC was calculated for each specie, y_true is 1 or 0 
+#' for pval < 0.01 or pval > 0.2, respeceively,
+#' y_pred is normalized alignment score (AS / 50).
+#'
+#' Our function works on a single sample.
+#' @importFrom PRROC roc.curve
```
Comment (zwdzwd) [inline R/species.r:14]: remove PRROC call. use some custom code like 
```
 labels <- as.logical(labels)
    n1 <- sum(labels)
    n2 <- sum(!labels)
    R1 <- sum(rank(scores)[labels])
    U1 <- R1 - n1 * (n1 + 1)/2
    U1/(n1 * n2)
```

File: R/species.r [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+#' @import sesameData
```
Comment (zwdzwd) [inline R/species.r:15]: You don't need to import sesameData. Depended package are imported automatically

File: R/species.r [hunk 3]
```diff
@@ -0,0 +1,114 @@
...
+#' @examples
```
Comment (zwdzwd) [inline R/species.r:16]: The current example is too complicated. Just call inferSpecies for one sample/sset is enough

File: R/species.r [hunk 4]
```diff
@@ -0,0 +1,114 @@
...
+#'    )
+#'
+#' write.table(as.data.frame(result),file="AUC_for_species.txt",sep='\t',quote=FALSE,col.names=T)
+#' @export
+
+inferSpecies <- function(sset,df_as=NULL,infer.human=T,topN=3000,ret.max=T) {
+    pvalue=pval(sset)
+    pvalue <- pvalue[intersect(names(pvalue),rownames(df_as))]
+    pos_probes=sort(pvalue[pvalue< 0.01],decreasing=F)
+    neg_probes=sort(pvalue[pvalue > 0.2],decreasing=T)
```
Comment (zwdzwd) [inline R/species.r:69]: we should make 0.01 and 0.2 arguments of the function

File: R/species.r [hunk 5]
```diff
@@ -0,0 +1,114 @@
...
+	    else {
+		    return(as.data.frame(t(sapply(colnames(df_as),function() {
+			    list(auc=NA,p.value=NA,species=NA,taxid=NA)
+		    }))))
+	    }
+    }
+        
+    
+    res = as.data.frame(t(sapply(colnames(df_as),function(s) {
+	    PRROC_obj <- roc.curve(scores.class0 = df_as[,s], weights.class0=y_true,curve=F)
```
Comment (zwdzwd) [inline R/species.r:98]: AUC is the same as wilcox test statistic. no need to calculate twice.

**Review by zwdzwd — CHANGES_REQUESTED** [review]
let's called the file species.R to be consistent with the other files.

File: R/species.r (hunk 2 above)
Comment (DingWB) [reply R/species.r]: It has been removed.

File: R/species.r (hunk 4 above)
Comment (DingWB) [reply R/species.r]: threshold_pos and threshold_neg have been added.

File: R/species.r (hunk 5 above)
Comment (DingWB) [reply R/species.r]: U1 eauqls to the wilcox test statistic, but when I canculated p-value with
`
z <- (R1 - n1*(n1+n2+1)/2) / sqrt(n1*n2*(n1+n2+1))
p <- pnorm(z, lower.tail=FALSE)
`
this p-value is different from wilcox test pvalue.

File: R/species.r (hunk 1 above)
Comment (DingWB) [reply R/species.r]: It has been modified.

File: R/species.R [hunk 6]
```diff
@@ -0,0 +1,87 @@
...
+#'      	res=inferSpecies(sset,df_as,ret.max=T)
+#'    },mc.cores=8)
+#'    )
+#' @export
+
+inferSpecies <- function(sset,df_as=NULL,topN=3000,
+			threshold.pos=0.01,threshold.neg=0.2,ret.max=T,
+			balance=T) {
+    if (is.null(df_as)) {
+    	df_as=sesameDataGet('df_as')
```
Comment (zwdzwd) [inline R/species.R:32]: I changed df_as to Mammal40.alignmentScore. Please test using sesameDataGet("Mammal40.alignmentScore") and update accordingly

**Review by zwdzwd — CHANGES_REQUESTED** [review]

File: R/Species.r [hunk 7]
```diff
@@ -0,0 +1,97 @@
...
+#' sset <- sesameDataGet('EPIC.1.LNCaP')$sset
+#' df_as <- sesameDataGet() #TODO
+#' inferSpecies(sset)
+#' 
+#' @examples
+#' library(tidyverse)
+#' library(sesame)
+#' library(parallel)
+#' library(readxl)
+#' indir="/mnt/isilon/zhou_lab/projects/20191212_GEO_datasets/IDATs/Mammal40/"
```
Comment (zwdzwd) [inline R/Species.r]: we cannot use these local paths

File: R/Species.r [hunk 8]
```diff
@@ -0,0 +1,97 @@
...
+#' @return species (NCBI official species names)
+#' We infer species based on probes pvalues and alignment score.
+#' We calculate AUC for each species, y_true is whether pval < 0.01
+#' y_pred is normalized alignment score (AS / 50).
+#'
+#' Our function works on a single sample.
+#' @importFrom PRROC roc.curve
+#' @import sesameData
+#' @examples
+#' sset <- sesameDataGet('EPIC.1.LNCaP')$sset
```
Comment (zwdzwd) [inline R/Species.r]: delete?

File: R/Species.r [hunk 9]
```diff
@@ -0,0 +1,97 @@
...
+#' @examples
+#' library(tidyverse)
+#' library(sesame)
+#' library(parallel)
+#' library(readxl)
+#' indir="/mnt/isilon/zhou_lab/projects/20191212_GEO_datasets/IDATs/Mammal40/"
+#' AS_dir="/mnt/isilon/zhoulab/labprojects/20200228_Mouse_Array_Project/20210324_SeSAme_Wubin/data/AS_A_probe_species_matrix.txt.gz"
+#' samplesheet="/mnt/isilon/zhoulab/labsamplesheets/2021/20210325_Mammal40_Merged_GEO_Steve_Wubin_samplesheets.xlsx"
+#' 
+#' sample_info=read_excel(samplesheet)
```
Comment (zwdzwd) [inline R/Species.r]: we should prepare a Mammal40 example dataset in sesameData

**Review by zwdzwd — COMMENTED** [review]
We need to rewrite the example

File: R/Species.r (hunk 7 above)
Comment (DingWB) [reply R/Species.r]: Which version were you looking at? This was the very old version.

**Pull request merged**